pretty_midi/pretty_midi.py

```diff
@@ -117,6 +146,38 @@ def _load_tempo_changes(self, midi_data):
                     if tick_scale != last_tick_scale:
                         self.__tick_scales.append((event.tick, tick_scale))
 
+    def _load_metadata(self, midi_data):
+        """Populates the lists of tempo key and time signature changes
+
+        Populates self.__time_signatures with TimeSignature objects.
+            It can be accessed by the get_time_signatures method.
+        Populates self.__key_changes with KeySignature objects.
+            It can be accessed by the get_key_signatures method.
+
+        Parameters
+        ----------
+            - midi_data : midi.FileReader
+                MIDI object from which data will be read
+        """
+
+        # list to store key signature changes
+        self.__key_changes = []
```

Comment:
I think they should be private.
If the midi file was written properly, that means the user shouldn't have a need to set the time nor key signatures.

If the midi file is a transcription from someone's performance on an instrument, then the user might have the need to change it. In this case, the user can use a setter method, which enforces the correct data structure, data types, etc...

What do you think?